src/photoshop.cpp

```diff
@@ -0,0 +1,182 @@
+#endif
+    // Pascal string is padded to have an even size (including size byte)
+    byte psSize = pPsData[position] + 1;
+    psSize += (psSize & 1);
+    position += psSize;
+    if (position + 4 > sizePsData) {
+#ifdef EXIV2_DEBUG_MESSAGES
+      std::cerr << "Warning: "
+                << "Invalid or extended Photoshop IRB\n";
+#endif
+      return -2;
+    }
+    uint32_t dataSize = getULong(pPsData + position, bigEndian);
+    position += 4;
+    if (dataSize > (sizePsData - position)) {
+#ifdef EXIV2_DEBUG_MESSAGES
+      std::cerr << "Warning: "
+                << "Invalid Photoshop IRB data size " << dataSize << " or extended Photoshop IRB\n";
+#endif
+      return -2;
+    }
+#ifdef EXIV2_DEBUG_MESSAGES
+    if ((dataSize & 1) && position + dataSize == sizePsData) {
+      std::cerr << "Warning: "
+                << "Photoshop IRB data is not padded to even size\n";
+    }
+#endif
+    if (type == psTag) {
+#ifdef EXIV2_DEBUG_MESSAGES
+      std::cerr << "ok\n";
+#endif
+      *sizeData = dataSize;
+      *sizeHdr = psSize + 10;
+      *record = hrd;
+      return 0;
+    }
+    // Data size is also padded to be even
+    position += dataSize + (dataSize & 1);
+  }
+#ifdef EXIV2_DEBUG_MESSAGES
+  std::cerr << "pPsData doesn't start with '8BIM'\n";
+#endif
+  if (position < sizePsData) {
+#ifdef EXIV2_DEBUG_MESSAGES
+    std::cerr << "Warning: "
+              << "Invalid or extended Photoshop IRB\n";
+#endif
+    return -2;
+  }
+  return 3;
+}
+
+int Photoshop::locateIptcIrb(const byte* pPsData, size_t sizePsData, const byte** record, uint32_t* const sizeHdr,
+                             uint32_t* const sizeData) {
+  return locateIrb(pPsData, sizePsData, iptc_, record, sizeHdr, sizeData);
+}
+
+int Photoshop::locatePreviewIrb(const byte* pPsData, size_t sizePsData, const byte** record, uint32_t* const sizeHdr,
+                                uint32_t* const sizeData) {
+  return locateIrb(pPsData, sizePsData, preview_, record, sizeHdr, sizeData);
+}
+
+DataBuf Photoshop::setIptcIrb(const byte* pPsData, size_t sizePsData, const IptcData& iptcData) {
+#ifdef EXIV2_DEBUG_MESSAGES
+  std::cerr << "IRB block at the beginning of Photoshop::setIptcIrb\n";
+  if (sizePsData == 0)
+    std::cerr << "  None.\n";
+  else
+    hexdump(std::cerr, pPsData, sizePsData);
+#endif
+  const byte* record = pPsData;
+  uint32_t sizeIptc = 0;
+  uint32_t sizeHdr = 0;
+  DataBuf rc;
+  if (0 > Photoshop::locateIptcIrb(pPsData, sizePsData, &record, &sizeHdr, &sizeIptc)) {
+    return rc;
+  }
+
+  Blob psBlob;
+  const auto sizeFront = static_cast<size_t>(record - pPsData);
+  // Write data before old record.
+  if (sizePsData > 0 && sizeFront > 0) {
+    append(psBlob, pPsData, sizeFront);
+  }
+
+  // Write new iptc record if we have it
+  DataBuf rawIptc = IptcParser::encode(iptcData);
+  if (!rawIptc.empty()) {
+    std::array<byte, 12> tmpBuf;
+    std::copy_n(Photoshop::irbId_[0], 4, tmpBuf.data());
```

Comment:
.begin() can be used to fix this warning by CodeQL

Reply:
I had to use `.front()` but yes, if we can avoid those warnings we should give it a try 😉